cpp/cmake/thirdparty/get_spdlog.cmake

```diff
@@ -0,0 +1,34 @@
+# =============================================================================
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+# Use CPM to find or clone speedlog
+function(find_and_configure_spdlog)
+
+  include(${rapids-cmake-dir}/cpm/spdlog.cmake)
+  rapids_cpm_spdlog(FMT_OPTION "EXTERNAL_FMT_HO" INSTALL_EXPORT_SET cudf-exports)
+  rapids_export_package(BUILD spdlog cudf-exports)
```

Comment:
@robertmaynard is this necessary (rather than specifying the BUILD_EXPORT_SET above) because spdlog isn't properly set up for CMake to have its build directories linked against via CPMFindPackage?

Reply:
I see that this is also what rmm is doing so I assume it is correct and won't block the PR on the question, still curious to hear from Rob for my edification.

Reply:
Correct currently spdlog doesn't provide a proper build directory config file. I have added a rapids-cmake issue ( https://github.com/rapidsai/rapids-cmake/issues/387 ) so that project don't need to do this logic themselves.

Reply:
Awesome thank you.